lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -684,6 +694,237 @@ public void resume_publish_queue() {
         }
     }
 
+    /**
+     * In case of resume success, verify that pending messages are resent. By blocking ack/nacks before sending the
+     * message while connected and then disconnect, add some mroe messages
+     * */
+    @Test
+    public void resume_publish_resend_pending_messages_when_resume_is_successful() {
+        Log.d(TAG, "resume_publish_resend_pending_messages: ");
+        final long delay = 200;
+        final String channelName = "resume_publish_queue";
+        AblyRealtime sender = null;
+        AblyRealtime receiver = null;
+        try {
+            final MockWebsocketFactory mockWebsocketFactory = new MockWebsocketFactory();
+
+            //block ack/nack messages to simulate pending message
+            DebugOptions senderOptions = createOptions(testVars.keys[0].keyStr);
+            senderOptions.transportFactory = mockWebsocketFactory;
+            sender = new AblyRealtime(senderOptions);
+            receiver = new AblyRealtime(senderOptions);
+            final Channel senderChannel = sender.channels.get(channelName);
+            senderChannel.attach();
+            (new ChannelWaiter(senderChannel)).waitFor(ChannelState.attached);
+            assertEquals(
+                "The sender's channel should be attached",
+                senderChannel.state, ChannelState.attached
+            );
+
+            /* create and attach channel to recv on */
+            final Channel receiverChannel = receiver.channels.get(channelName);
+            receiverChannel.attach();
+            (new ChannelWaiter(receiverChannel)).waitFor(ChannelState.attached);
+            assertEquals(
+                "The receiver's channel should be attached",
+                receiverChannel.state, ChannelState.attached
+            );
+            /* subscribe */
+            MessageWaiter messageWaiter =  new MessageWaiter(receiverChannel);
+            //block acks nacks
+            mockWebsocketFactory.blockReceive(message -> message.action == ProtocolMessage.Action.ack ||
+                message.action == ProtocolMessage.Action.nack);
+            final int messageCount = 6; // make 3 pending 3 queued
```

Comment:
Personally I think I would have had `messageCountPerPhase = 3`

Reply:
In most recent changes, I removed this and just explicitly set the constant 3 for each send as I thought this was a bit confusing. I hope that's ok.